src/main/java/org/springframework/integration/kafka/outbound/KafkaProducerMessageHandler.java

```diff
@@ -108,7 +112,9 @@ protected void handleMessageInternal(final Message<?> message) throws Exception
 				? this.messageKeyExpression.getValue(this.evaluationContext, message)
 				: message.getHeaders().get(KafkaHeaders.MESSAGE_KEY);
 
-		this.kafkaProducerContext.send(topic, partitionId, messageKey, message.getPayload());
+		Future<RecordMetadata> future = this.kafkaProducerContext.send(topic, partitionId, messageKey,
```

Comment:
We shouldn't do this here directly, because it completely overrides the default asynchronous behaviour of the post-0.8.2 Kafka producer. 

The current `ProducerContext` already supports an option for a synchronous producer which displays the expected behaviour , via https://github.com/spring-projects/spring-integration-kafka/blob/master/src/main/java/org/springframework/integration/kafka/support/ProducerConfiguration.java#L93, with the XML configuration option here: https://github.com/spring-projects/spring-integration-kafka/blob/master/src/main/resources/org/springframework/integration/kafka/config/spring-integration-kafka-2.0.xsd#L237

I believe that for 2.0 this can be reworked as an option of an largely enhanced KafkaTemplate, and for 1.3 we can rely on the current support.